### ISSUE TITLE
Feature request for menu bar item with search field

### ISSUE THREAD

**Issue by hjchanna** [MacPass/MacPass#1170]
First of all, thank you very much everyone's effort to build such a great tool ! 💯 I am a MacPass user since a long time, and it always being useful to me.

I always find it difficult to switch Mac Pass to grab a url, username or a password. 

I would love to see a menu bar icon for Mac Pass which opens a simple dialog box (like a context menu) next to it. The dialog box would have a search search field that would help to quickly find whatever items in the default database. The workflow could be more efficient when it is supported with shortcut keys such as a shortcut key to open the dialog box with focus to the search field and return key would copy the url, username or password to the clipboard. Or even pass copy-paste key strokes to directly enter to the selected text field in the external application.

I can see such a feature is in place for [SnipptsLib](https://www.renfei.org/snippets-lab/) and I have attached the screenshot of that.

![](https://www.renfei.org/snippets-lab/images/Landing/assistant-new.jpg)

Thank you!

**Comment by georgesnow** [comment]
Just as note I messed around with this on my plug-in:

https://github.com/georgesnow/MacPassRevealer/tree/popover

I never was able to take the Macpass's current view controllers and instiate them properly into the popover. I only could get password gen or only a list view￼. I believe cause MP is using NSDocument. 

That's all to say I believe the GUI would need to be built from scratch. I love this idea as this was originally purpose of my plug-in.

**Comment by jpransxdev** [comment]
> Just as note I messed around with this on my plug-in:
> 
> https://github.com/georgesnow/MacPassRevealer/tree/popover
> 
> I never was able to take the Macpass's current view controllers and instiate them properly into the popover. I only could get password gen or only a list view￼. I believe cause MP is using NSDocument.
> 
> That's all to say I believe the GUI would need to be built from scratch. I love this idea as this was originally purpose of my plug-in.

Thanks for the plugin btw! I'm using it right now, it's not perfect but it's something :) Main reason why I was able to switch back back from Keepass. I really hate electron apps...

**Comment by georgesnow** [comment]
Thanks! Been. A little neglected. I did have 2 false starts of trying to port the code over the main app. However the near term I do want to clean up the code and reliability.

**Comment by mstarke** [comment]
It should be rather trivial to get a plugin to retrieve all the open NSDocuments and their trees. Thus one could then search in them and show a list inside the menu popup. I'm unsure how well this would work but it might be worth a try. I did take a look at your code @georgesnow and I think it might be better to start this separate UI from scratch. Even as a plugin for starters and then we could fold it into MacPass if everyone is happy with it.

**Comment by georgesnow** [comment]
Agreed. The only thing is UI is not my strong point 😬. That's why I had started down the road of using the existing views. I am more than willing to contribute to the code, but if there is person who can flesh  out the layout that would be huge. Otherwise ￼it will take me A lot longer to implement￼￼
